lib/embed-docs/embedDocs.ts

```diff
@@ -279,3 +252,77 @@ export function splitTextByHeaders(
 
   return results;
 }
+
+export function splitIntoTextChunks(
+  sectionText: string,
+  ignoreLines: string[],
+): string[] {
+  return joinShortChunks(
+    sectionText
+      .split("\n")
+      .filter((chunk) => isTextWithSubstance(chunk))
+      // Don't just restate the title
+      .filter((chunk) => !ignoreLines.includes(RemoveMarkdown(chunk)))
+      // Tabs are more token-efficient than 4 spaces
+      .map((chunk) => chunk.replaceAll(/ {4}/g, "\t") + "\n")
+      .map((chunk) => {
+        // Break up large paragraphs into sentences
+        const chunkTokenCount = getTokenCount(chunk);
+        if (chunkTokenCount <= 80) return chunk;
+        return splitMarkdownIntoSentences(chunk);
+      })
+      .flat(),
+    25,
+  );
+}
+
+export function splitMarkdownIntoSentences(markdown: string) {
+  let tmpMarkedItems: string[] = [];
+  let tmpString = markdown;
+
```

Comment:
what's the behaviour of this badboy if it encounters headers?

Reply:
Great question. It doesn't do anything to headers.

However, the short answer is really that it won't come across headers unless a header is longer than 80 tokens

This is because we're splitting by lines above